add CI options and utils

shunjizhan opened the pull request:
similar to our previous waffle examples https://github.com/AcalaNetwork/bodhi.js/tree/master/examples/waffle, we need to add ability to switch between using local (default) url or CI url.

This commit copied the helper util functions from waffle examples to changed waffle tests accordingly.

shunjizhan committed:
add CI options and utils

shunjizhan commented on utils.ts:
```diff
@@ -0,0 +1,51 @@
+  return provider;
+};
+
+export const setup = async (urlOverwrite?: string) => {
```
this method is not used now, but is used by old waffle examples, so might be useful in the future for new waffle tutorials

ThunderDeliverer commented:
This is great! 👌  Before we merge this PR, I need to document the changes and update the tutorials, so don't merge it just yet.

shunjizhan committed:
fix NFT folder name

shunjizhan commented:
I think we are ready to merge this? It's utils can be reused by #3

ThunderDeliverer commented:
Let me document it today, I'll merge it and add the helper to the other tutorials

ThunderDeliverer commented:
@shunjizhan why don't we use the https://github.com/AcalaNetwork/waffle-tutorials/blob/6928e9b885febd073879e1ed6b8b7e6a339da126/hello-world/src/setup.ts as it already supports the `URL` and `SEED` override?

shunjizhan commented:
yeah we actually could, `setup` is like a superset of `getTestProvider`,  I didn't choose to separate them, just followed what the original waffle examples did, and factored them out.

Do you prefer we use setup for all? I am fine with either way

ThunderDeliverer commented:
Yeah, lets only have one way of setting up the `provider` and `wallets`. I'll move the `setup.ts` to `utils` folder and rewrite the relevant parts of the tutorials.

ThunderDeliverer committed:
Added utils to HelloWorld

utils/ folder was added it includes the setup to set the wallet and
provider for the deploy script as well as test provider for the tests.
This change allows us to run the tutorial in the CI while it still being
treated as it's own self-contained project.

ThunderDeliverer committed:
Added utils to Echo

utils/ folder was added it includes the setup to set the wallet and
provider for the deploy script as well as test provider for the tests.
This change allows us to run the tutorial in the CI while it still being
treated as it's own self-contained project.

ThunderDeliverer committed:
Added utils to Token

utils/ folder was added it includes the setup to set the wallet and
provider for the deploy script as well as test provider for the tests.
This change allows us to run the tutorial in the CI while it still being
treated as it's own self-contained project.

ThunderDeliverer committed:
Added utils to NFT

utils/ folder was added it includes the setup to set the wallet and
provider for the deploy script as well as test provider for the tests.
This change allows us to run the tutorial in the CI while it still being
treated as it's own self-contained project.

ThunderDeliverer committed:
Removed utils from root

Removed general utils as all of the tutorials received their own utility
directory and use it as self-contained projects.

ThunderDeliverer commented:
@shunjizhan I migrated general utility file into separate utility folders within each tutorial. Take a look and if you have no comments, we can merge this. :raised_hands:

shunjizhan commented on utils.ts:
```diff
@@ -0,0 +1,51 @@
+import { TestProvider, TestAccountSigningKey, Provider, Signer } from '@acala-network/bodhi';
```
so we don't need this file anymore, and can remove it?

ThunderDeliverer replied:
I was certain that I pushed the commit that removed this file. It's up now :see_no_evil:

ThunderDeliverer reviewed (approved):
Let's do this! :rocket:
